doc/01_general/04_architecture.md

```diff
@@ -0,0 +1,319 @@
+- ```gnss``` ([sensor_msgs/NavSatFix](https://carla.readthedocs.io/projects/ros-bridge/en/latest/ros_sensors/))
+
+Publishes:
+
+- ```traffic_lights``` (Custom msg:
+  state ([std_msgs/UInt8 Message](https://docs.ros.org/en/api/std_msgs/html/msg/UInt8.html])),
+  position ([geometry_msgs/Pose Message](http://docs.ros.org/en/noetic/api/geometry_msgs/html/msg/Pose.html)),
+  distance_to_stop_line ([std_msgs/Float64 Message](http://docs.ros.org/en/api/std_msgs/html/msg/Float64.html)))
+
+### Traffic Signs Detection
+
+Recognizes traffic signs.
+In particular traffic signs that are relevant for the correct traffic behavior of the ego vehicle,
+are recognized early and reliably.
+
+Subscriptions:
+
+- ```rgb_camera``` ([sensor_msgs/Image](https://docs.ros.org/en/api/sensor_msgs/html/msg/Image.html))
+- ```lidar``` ([sensor_msgs/PointCloud2](https://docs.ros.org/en/api/sensor_msgs/html/msg/PointCloud2.html))
+
+Publishes:
+
+- ```traffic_signs``` (Custom msg:
+  type ([std_msgs/UInt8 Message](https://docs.ros.org/en/api/std_msgs/html/msg/UInt8.html])),
+  position ([geometry_msgs/Pose Message](http://docs.ros.org/en/noetic/api/geometry_msgs/html/msg/Pose.html)),
+  distance_to_stop_line
+  distance_to_stop_line ([std_msgs/Float64 Message](http://docs.ros.org/en/api/std_msgs/html/msg/Float64.html)))
+
+### Prediction
+
+Tries to predict the movement of dynamic objects recognized in the
+[Obstacle Detection and Classification](#Obstacle-Detection-and-Classification).
+
+Subscriptions:
+
+- ```obstacles``` ([vision_msgs/Detection3DArray Message](http://docs.ros.org/en/api/vision_msgs/html/msg/Detection3DArray.html))
+- ```lanes``` ([derived_object_msgs/LaneModels Message](http://docs.ros.org/en/kinetic/api/derived_object_msgs/html/msg/LaneModels.html))
+
+Publishes:
+
+- ```predictions``` (Custom msg: array of
+  vehicle_id ([std_msgs/UInt16 Message](https://docs.ros.org/en/api/std_msgs/html/msg/UInt16.html)) and
+  vehicle_path ([nav_msgs/Path Message](http://docs.ros.org/en/noetic/api/nav_msgs/html/msg/Path.html)))
+
+### Localization
+
+Provides corrected accurate position, direction and speed of the ego vehicle
+
+Subscriptions:
+
+- ```map``` ([std_msgs/String](https://docs.ros.org/en/api/std_msgs/html/msg/String.html))
+- ```imu``` ([sensor_msgs/Imu Message](https://docs.ros.org/en/api/sensor_msgs/html/msg/Imu.html))
+- ```speedometer``` ([std_msgs/Float32](https://docs.ros.org/en/api/std_msgs/html/msg/Float32.html))
+- ```gnss``` ([sensor_msgs/NavSatFix](https://carla.readthedocs.io/projects/ros-bridge/en/latest/ros_sensors/))
+
+Publishes:
+
+- ```ego_position``` ([nav_msgs/Odometry Message](http://docs.ros.org/en/noetic/api/nav_msgs/html/msg/Odometry.html))
+
+## Planning
+
+The planning uses the data from the [Perception](#Perception) to find a path on which the ego vehicle can safely reach
+its destination
+
+Further information can be found [here](../03_research/03_planning/Readme.md).
```

Comment:
Again, maybe "Further information regarding the perception can be found..." would be more precise.

Reply:
Fixed in 3f8ef00137e56173f50a933439c7fa21ba3738a6